src/ViewModel/Factory/SiteHeaderFactory.php

```diff
@@ -117,11 +119,15 @@ public function createSiteHeader(Model $item = null, Profile $profile = null) :
             $searchItem,
         ]);
 
+        if ($this->authorizationChecker->isGranted('FEATURE_XPUB')) {
+            $submitUrl = $this->urlGenerator->generate('submit');
+        }
+
         $secondaryLinks = [
             NavLinkedItem::asLink(new Link('About', $this->urlGenerator->generate('about'))),
             NavLinkedItem::asLink(new Link('Community', $this->urlGenerator->generate('community'))),
             NavLinkedItem::asButton(
-                Button::link('Submit my research', 'http://submit.elifesciences.org/', Button::SIZE_EXTRA_SMALL)
+                Button::link('Submit my research', $submitUrl ?? $this->submitUrl, Button::SIZE_EXTRA_SMALL)
```

Comment:
Should keep hard-coded value when feature flag is off? See https://github.com/elifesciences/builder-configuration/pull/8#discussion_r222622085.

Reply:
From the conversation we had with @BlueReZZ ... I seem to remember that the "switching" will be done at the proxy level in nginx (on `submit.elifesciences.org`), not at the application level. So not sure if we need this feature? Apologies if this didn't get relayed to you Chris! Worth checking though with Paul and/or Hannah

Reply:
We won't want it hard-coded to allow using different environments.